### DISCUSSION TITLE
Is my Cart Reader salvageable?

### DISCUSSION THREAD

**Discussion by raylaxing** [sanni/cartreader#381] [Need help]
Hi there!
I've been using the Cart reader for years now, it's been excellent! The years of wear and tear were too much for it, resulting in the left button slowly breaking over time. (This is a V3 model). I replaced the button today, however, I must have done some damage in removing the broken button as the new button now acts like the the button to the right (both buttons are now acting as enter!).  Do you guys think this is salvageable, or should I just start all over again? I can imagine there might be some sort of trace issue caused by my iron, but I'm not too sure what I could do to rectify this. Is there a PC controlled feature? (I saw there's a Serial output feature, does a serial input feature exist? ) 

Thanks!

**Comment by Dakkaron** [comment]
Did you flash the firmware while fixing?

Do you have pictures of the solder job?

Serial output is serial input. To enable it, open options.h. In that file, remove the leading // in front of the line `#define SERIAL_MONITOR`.

Buttons and display will now not work anymore.

Connect the Arduino to your PC using USB and then open any serial monitor of your choice (e.g. the one in the Arduino IDE). Baudrate is 9600.

**Reply by raylaxing** [reply]
Thanks for the response! 
I actually stumbled on this just before you responded- I have just uploaded the latest firmware (from a very, very old firmware!) and enabled serial monitor and it's all working great! This feels like a much better option than using the two physical buttons anyway :)

**Comment by sanni** [comment]
The first button acts the same as the second button if it's pressed longer. 
That means that either the button is broken or you shorted the button pin to GND, remember the whole casing of the button is GND and when you solder the tiny leg none of the solder is allowed to touch the case of the button.
